### PR TITLE
Hubs display as closed when not configured for payment or shipping

### PR DESCRIPTION
**Pull request by Matt-Yorkley**
Addressing issue #1343 (and duplicate issue #495 ?)

Hubs now display as closed when browsing shops in the frontend if they are not "ready for checkout", i.e. don't have payment or shipping methods properly configured.

Previously they displayed as open, allowed a user to shop and add things to the basket, but then bounced the user to the homepage with an error message and emptied the cart once the user tried to go to the checkout.

The backend already displays messages to an enterprise user if their enterprise does not have payment or shipping methods properly configured, and correctly warns that customers will not be able to shop at their hub:

screenshot from 2016-12-20 12-37-17

Updated specs to reflect this.

### CODE REVIEW

File: lib/open_food_network/enterprise_injection_data.rb [hunk 1]
```diff
@@ -1,7 +1,7 @@
 module OpenFoodNetwork
   class EnterpriseInjectionData
     def active_distributors
-      @active_distributors ||= Enterprise.distributors_with_active_order_cycles
+      @active_distributors ||= Enterprise.distributors_with_active_order_cycles.andand.ready_for_checkout
```
Comment (mkllnk) [inline lib/open_food_network/enterprise_injection_data.rb]: Since `distributors_with_active_order_cycles` is a scope, I don't think you need `andand`.

Comment (mkllnk) [inline lib/open_food_network/enterprise_injection_data.rb]: I found three usages of this:

- app/helpers/shared_helper.rb
- app/serializers/api/enterprise_serializer.rb
- app/views/json/partials/_hub.rabl

All of them call the `include?` method on that collection to know if a certain enterprise is active. That sounds like a really poor way of getting that attribute. The computation time will grow linearly with the number of active enterprises.

So this fixes the issue, but it also points to a possible performance problem.

File: spec/features/consumer/shops_spec.rb [hunk 2]
```diff
@@ -201,4 +201,3 @@ def click_link_and_ensure(link_text, check)
     end
   end
 end
-
```
Comment (mkllnk) [inline spec/features/consumer/shops_spec.rb]: Is this to trigger a Buildkite build or Travis? It seems pretty ugly, but Buildkite has no proper search function for builds. Maybe that is something to ask them about. In Aus we tend to rebase branches.

Comment (lin-d-hop) [inline spec/features/consumer/shops_spec.rb]: Ok great I'll rebase in future.
Yeah was purely to force a buildkite build. Agreed it's ugly!

**Comment by daniellemoorhead** [comment]
@Matt-Yorkley can we close this one now that it's been rebased as #1516?

**Comment by Matt-Yorkley** [comment]
@daniellemoorhead Ah, I'd already rebased this branch and pushed new commits on top of it. Also I didn't create the other rebased one, so I can't push directly to it. I've cleaned it up and resubmitted as #1522.

**Pull request closed without merging**